src/rgw/rgw_sync_module_aws.cc

```diff
@@ -811,7 +811,7 @@ class RGWAWSHandleRemoteObjCBCR: public RGWStatRemoteObjCBCR {
                                                   sync_env->http_manager,
                                                   target_bucket_name, nullptr, bl, nullptr));
         }
-        if (retcode < 0) {
+        if (retcode < 0 && retcode != -ENOLCK) {
```

Comment:
This doesn't sound right to me. Where do you get the `-ENOLCK`, and why is ignoring it here fixes it for you?

Reply:
sorry, that's my mistake, it should be `-ENOTEMPTY` in this case.